File: app/representations/api/widget.rb
module Api::Widget
  extend ActiveSupport::Concern

  included do

    api_accessible :api_simple do |t|
      t.add :id
      t.add :api_key
      t.add :height
      t.add :width
      t.add :lat
      t.add :lon
      t.add :user_id
      t.add :categories
    end

  end
end

File: app/assets/javascripts/WidgetBuilder/models/Widget.js
import { Record, List } from 'immutable';

const Widget = new Record({
  id: null,
  apiKey: null,
  height: 500,
  width: 400,
  lat: 52.50521,
  lon: 13.4231,
  userId: null,
  categories: false,
  boundingBox: null,
  src: null,
  providers: new List()
});

export default Widget;
